freud/util.pyx

```diff
@@ -212,8 +212,12 @@ def _convert_array(array, shape=None, dtype=np.float32):
     Returns:
         :class:`numpy.ndarray`: Array.
     """
+    if array is None:
+        return np.empty(shape, dtype=dtype)
+
     array = np.asarray(array)
     return_arr = np.require(array, dtype=dtype, requirements=['C'])
```

Comment:
I would try to combine these steps so that the case where `array is None` doesn't have to _also_ convert to an array and require C-ordering / contiguity. Basically just do something like:
```python
if array is None:
    # Construct a valid return_array that meets the array requirements.
else:
    # Perform the "asarray" and enforce array requirements.
```

Reply:
I'm not sure I understand why cases where ```array is None``` also has to convert I thought what we're doing here is that we generate an empty array that has the given shape and type if ```array is None```, and we just skip the converting part?

Reply:
That's my mistake, I read the code incorrectly. I saw `return_arr = np.empty(shape, dtype=dtype)` instead of `return np.empty(shape, dtype=dtype)`.